### PR TITLE
Travis

### PR DESCRIPTION
**Pull request by gidden**
updates for travis as a new ci platform. this PR is known to break mac-based polyphemus CI due to travis selector usage.

### CODE REVIEW

**Comment by scopatz** [comment]
Hi @gidden - Do you have working run you can point me to?

**Comment by gidden** [comment]
https://github.com/gidden/cyclus/pull/16 is being tuned up

On Thu, Apr 23, 2015 at 11:59 AM, Anthony Scopatz <notifications@github.com>
wrote:

> Hi @gidden <https://github.com/gidden> - Do you have working run you can
> point me to?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/ciclus/pull/62#issuecomment-95653585>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

File: cycamore/build.sh [hunk 1]
```diff
@@ -1,18 +1,30 @@
 #!/bin/bash
 
-mkdir build; cd build;
+mkdir build; 
+cd build;
```
Comment (scopatz) [inline cycamore/build.sh]: Weird to have trailing semicolons.

File: cycamore/build.sh [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
 export MACOSX_DEPLOYMENT_TARGET=
+
+# # hack fix for hdf5 issues
+# if [[  `uname` == 'Linux' ]]; then
+#   ln -s $PREFIX/lib/libhdf5.so.9 $PREFIX/lib/libhdf5.so.8
+#   ln -s $PREFIX/lib/libhdf5_hl.so.9 $PREFIX/lib/libhdf5_hl.so.8
+# else
+#   ln -s $PREFIX/lib/libhdf5.9.dylib $PREFIX/lib/libhdf5.8.dylib
+#   ln -s $PREFIX/lib/libhdf5_hl.9.dylib $PREFIX/lib/libhdf5_hl.8.dylib
+# fi
```
Comment (scopatz) [inline cycamore/build.sh]: Remove comments.

File: cycamore/build.sh [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
+
 $PREFIX/bin/cyclus --version
+
 if [[  `uname` == 'Linux' ]]; then
 
-    cmake ..  -DCMAKE_INSTALL_PREFIX=$PREFIX -DCYCLUS_ROOT_DIR=$PREFIX -DBOOST_ROOT=$PREFIX  -DBOOST_LIBRARYDIR=$PREFIX/lib -DBoost_NO_SYSTEM_PATHS=ON -DCMAKE_BUILD_TYPE=Release -DLAPACK_LIBRARIES=$PREFIX/lib/liblapack.so -DBLAS_LIBRARIES=$PREFIX/lib/libblas.so
+    cmake ..  -DCMAKE_INSTALL_PREFIX=$PREFIX -DCYCLUS_ROOT_DIR=$PREFIX -DHDF5_ROOT=$PREFIX -DBOOST_ROOT=$PREFIX  -DBOOST_LIBRARYDIR=$PREFIX/lib -DBoost_NO_SYSTEM_PATHS=ON -DCMAKE_BUILD_TYPE=Release -DLAPACK_LIBRARIES=$PREFIX/lib/liblapack.so -DBLAS_LIBRARIES=$PREFIX/lib/libblas.so
```
Comment (scopatz) [inline cycamore/build.sh]: line breaks would be nice :)

Comment (gidden) [inline cycamore/build.sh]: I explicitly left comments for whomever comes next with the hdf5 issue. If
you really want them gone, I can remove them, but I think the possible
future help more than makes up for any detraction.

On Thu, Apr 23, 2015 at 12:28 PM, Anthony Scopatz <notifications@github.com>
wrote:

> In cycamore/build.sh
> <https://github.com/cyclus/ciclus/pull/62#discussion_r28983940>:
>
> >  if [[  `uname` == 'Linux' ]]; then
> >
> > -    cmake ..  -DCMAKE_INSTALL_PREFIX=$PREFIX -DCYCLUS_ROOT_DIR=$PREFIX -DBOOST_ROOT=$PREFIX  -DBOOST_LIBRARYDIR=$PREFIX/lib -DBoost_NO_SYSTEM_PATHS=ON -DCMAKE_BUILD_TYPE=Release -DLAPACK_LIBRARIES=$PREFIX/lib/liblapack.so -DBLAS_LIBRARIES=$PREFIX/lib/libblas.so
> > +    cmake ..  -DCMAKE_INSTALL_PREFIX=$PREFIX -DCYCLUS_ROOT_DIR=$PREFIX -DHDF5_ROOT=$PREFIX -DBOOST_ROOT=$PREFIX  -DBOOST_LIBRARYDIR=$PREFIX/lib -DBoost_NO_SYSTEM_PATHS=ON -DCMAKE_BUILD_TYPE=Release -DLAPACK_LIBRARIES=$PREFIX/lib/liblapack.so -DBLAS_LIBRARIES=$PREFIX/lib/libblas.so
>
> line breaks would be nice :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/ciclus/pull/62/files#r28983940>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

File: cyclus-deps/build.sh [hunk 4]
```diff
@@ -1,2 +1,12 @@
 #!/bin/bash
+
+# # hack fix for hdf5 issues
+# if [[ "${UNAME}" == 'Linux' ]]; then
+#   ln -s $PREFIX/lib/libhdf5.so.9 $PREFIX/lib/libhdf5.so.8
+#   ln -s $PREFIX/lib/libhdf5_hl.so.9 $PREFIX/lib/libhdf5_hl.so.8
+# else
+#   ln -s $PREFIX/lib/libhdf5.9.dylib $PREFIX/lib/libhdf5.8.dylib
+#   ln -s $PREFIX/lib/libhdf5_hl.9.dylib $PREFIX/lib/libhdf5_hl.8.dylib
+# fi
```
Comment (scopatz) [inline cyclus-deps/build.sh]: Is this file valuable?

File: cyclus/build.sh [hunk 5]
```diff
@@ -5,20 +5,32 @@ cd build
...
-else
 
+# # hack fix for hdf5 issues
+# if [[  `uname` == 'Linux' ]]; then
+#   ln -s $PREFIX/lib/libhdf5.so.9 $PREFIX/lib/libhdf5.so.8
+#   ln -s $PREFIX/lib/libhdf5_hl.so.9 $PREFIX/lib/libhdf5_hl.so.8
+# else
+#   ln -s $PREFIX/lib/libhdf5.9.dylib $PREFIX/lib/libhdf5.8.dylib
+#   ln -s $PREFIX/lib/libhdf5_hl.9.dylib $PREFIX/lib/libhdf5_hl.8.dylib
+# fi
```
Comment (scopatz) [inline cyclus/build.sh]: Remove comments

File: cyclus/post-link.sh [hunk 6]
```diff
@@ -1,4 +1,14 @@
...
+echo "post-link.sh, PREFIX: $PREFIX"
+
+# # hack fix for hdf5 issues
+# if [[  `uname` == 'Linux' ]]; then
+#   ln -s $PREFIX/lib/libhdf5.so.9 $PREFIX/lib/libhdf5.so.8
+#   ln -s $PREFIX/lib/libhdf5_hl.so.9 $PREFIX/lib/libhdf5_hl.so.8
+# else
+#   ln -s $PREFIX/lib/libhdf5.9.dylib $PREFIX/lib/libhdf5.8.dylib
+#   ln -s $PREFIX/lib/libhdf5_hl.9.dylib $PREFIX/lib/libhdf5_hl.8.dylib
+# fi
```
Comment (scopatz) [inline cyclus/post-link.sh]: remove comments

File: cyclus/build.sh [hunk 7]
```diff
@@ -5,20 +5,32 @@ cd build
...
+    cmake ..  -DCMAKE_INSTALL_PREFIX=$PREFIX -DHDF5_ROOT=$PREFIX -DBOOST_ROOT=$PREFIX  -DBOOST_LIBRARYDIR=$PREFIX/lib -DBoost_NO_SYSTEM_PATHS=ON -DCMAKE_BUILD_TYPE=Release -DLAPACK_LIBRARIES=$PREFIX/lib/liblapack.so -DBLAS_LIBRARIES=$PREFIX/lib/libblas.so
+else
+    echo $CFLAGS
+    echo $LDFLAGS
+    export MACOSX_DEPLOYMENT_TARGET=
+    export DYLD_LIBRARY_PATH=$PREFIX/lib
+    export LDFLAGS="-headerpad_max_install_names -headerpad"
+    export CFLAGS="-headerpad_max_install_names -headerpad"
+    export CXXFLAGS=
+    cmake ..  -DCMAKE_INSTALL_PREFIX=$PREFIX -DHDF5_ROOT=$PREFIX -DCOIN_ROOT_DIR=$PREFIX -DBOOST_ROOT=$PREFIX -DCMAKE_BUILD_TYPE=Release  -DLAPACK_LIBRARIES=$PREFIX/lib/liblapack.dylib -DBLAS_LIBRARIES=$PREFIX/lib/libblas.dylib
```
Comment (scopatz) [inline cyclus/build.sh]: line break.

File: cyclus/build.sh [hunk 8]
```diff
@@ -5,20 +5,32 @@ cd build
...
-echo $CFLAGS
-echo $LDFLAGS
-export MACOSX_DEPLOYMENT_TARGET=
-export DYLD_LIBRARY_PATH=$PREFIX/lib
-export LDFLAGS="-headerpad_max_install_names -headerpad"
-export CFLAGS="-headerpad_max_install_names -headerpad"
-export CXXFLAGS=
-    cmake ..  -DCMAKE_INSTALL_PREFIX=$PREFIX  -DCOIN_ROOT_DIR=$PREFIX -DBOOST_ROOT=$PREFIX -DCMAKE_BUILD_TYPE=Release  -DLAPACK_LIBRARIES=$PREFIX/lib/liblapack.dylib -DBLAS_LIBRARIES=$PREFIX/lib/libblas.dylib
+if [[  `uname` == 'Linux' ]]; then
+    cmake ..  -DCMAKE_INSTALL_PREFIX=$PREFIX -DHDF5_ROOT=$PREFIX -DBOOST_ROOT=$PREFIX  -DBOOST_LIBRARYDIR=$PREFIX/lib -DBoost_NO_SYSTEM_PATHS=ON -DCMAKE_BUILD_TYPE=Release -DLAPACK_LIBRARIES=$PREFIX/lib/liblapack.so -DBLAS_LIBRARIES=$PREFIX/lib/libblas.so
```
Comment (scopatz) [inline cyclus/build.sh]: line break

**Comment by scopatz** [comment]
Some style issue, but it looks like it passes.

**Comment by scopatz** [comment]
Thanks Matt!

File: cycamore/build.sh (hunk 2 above)
Comment (gidden) [inline cycamore/build.sh]: email didn't reply quite right --
I explicitly left comments for whomever comes next with the hdf5 issue. If
you really want them gone, I can remove them, but I think the possible
future help more than makes up for any detraction.

File: cyclus-deps/build.sh (hunk 4 above)
Comment (gidden) [inline cyclus-deps/build.sh]: I did not test the conda-deps build. That is likely off topic for this PR.

File: cycamore/build.sh (hunk 2 above)
Comment (scopatz) [inline cycamore/build.sh]: I think it is more confusing than anything. If you indented them for as dev docs they should only appear once and with a much clearer explanation of what they are there for.

File: cyclus-deps/build.sh (hunk 4 above)
Comment (scopatz) [inline cyclus-deps/build.sh]: Then you should probably not change the file :)

**Comment by gidden** [comment]
ping

**Comment by scopatz** [comment]
Still have the trailing semicolons.

**Comment by scopatz** [comment]
Otherwise looks good.  Can you get a new example run once the semcolons are taken care of please? Thanks

**Comment by gidden** [comment]
https://travis-ci.org/gidden/cyclus/builds/59773323

On Thu, Apr 23, 2015 at 2:02 PM, Anthony Scopatz <notifications@github.com>
wrote:

> Otherwise looks good. Can you get a new example run once the semcolons are
> taken care of please? Thanks
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cyclus/ciclus/pull/62#issuecomment-95686691>.
>



-- 
Matthew Gidden, Ph.D.
Postdoctoral Associate, Nuclear Engineering
The University of Wisconsin -- Madison
Ph. 225.892.3192

**Comment by scopatz** [comment]
Thanks @gidden

**Pull request merged**